actors/datacap/src/lib.rs

```diff
@@ -176,8 +178,8 @@ impl Actor {
             .context("state transaction failed")?;
 
         let mut st: State = rt.state()?;
-        let msg = Messenger { rt };
-        let intermediate = hook.call(&&msg).actor_result()?;
+        let msg = SyscallProvider { rt };
+        let intermediate = hook.call(as_token(&mut st, &msg).runtime()).actor_result()?;
```

Comment:
This is even more funky now as there's no logical need for the state to be involved in getting a runtime abstraction. You actually just need an `ActorRuntime`, as constructed in `as_token`. Copy just that bit of construction here?